pkg/apis/v1alpha1/nodeoverlay_validation.go

```diff
@@ -0,0 +1,132 @@
+	if err := validateWellKnownValues(ctx, requirement); err != nil {
+		errs = multierr.Append(errs, err)
+	}
+	for _, err := range validation.IsQualifiedName(requirement.Key) {
+		errs = multierr.Append(errs, fmt.Errorf("key %s is not a qualified name, %s", requirement.Key, err))
+	}
+	for _, value := range requirement.Values {
+		for _, err := range validation.IsValidLabelValue(value) {
+			errs = multierr.Append(errs, fmt.Errorf("invalid value %s for key %s, %s", value, requirement.Key, err))
+		}
+	}
+	if requirement.Operator == corev1.NodeSelectorOpIn && len(requirement.Values) == 0 {
+		errs = multierr.Append(errs, fmt.Errorf("key %s with operator %s must have a value defined", requirement.Key, requirement.Operator))
+	}
+
+	if requirement.Operator == corev1.NodeSelectorOpGt || requirement.Operator == corev1.NodeSelectorOpLt {
+		if len(requirement.Values) != 1 {
+			errs = multierr.Append(errs, fmt.Errorf("key %s with operator %s must have a single positive integer value", requirement.Key, requirement.Operator))
+		} else {
+			value, err := strconv.Atoi(requirement.Values[0])
+			if err != nil || value < 0 {
+				errs = multierr.Append(errs, fmt.Errorf("key %s with operator %s must have a single positive integer value", requirement.Key, requirement.Operator))
+			}
+		}
+	}
+	return errs
+}
+
+// ValidateWellKnownValues checks if the requirement has well known values.
+// An error will cause a NodePool's Readiness to transition to False.
+// It returns an error if all values are invalid.
+// It returns an error if there are not enough valid values to satisfy min values for a requirement with known values.
+// It logs if invalid values are found but valid values can be used.
+func validateWellKnownValues(ctx context.Context, requirement corev1.NodeSelectorRequirement) error {
+	// If the key doesn't have well-known values or the operator is not In, nothing to validate
+	if !v1.WellKnownLabels.Has(requirement.Key) || requirement.Operator != corev1.NodeSelectorOpIn {
+		return nil
+	}
+
+	// If the key doesn't have well-known values defined, nothing to validate
+	knownValues, exists := v1.WellKnownValuesForRequirements[requirement.Key]
+	if !exists {
+		return nil
+	}
+
+	values, invalidValues := lo.FilterReject(requirement.Values, func(val string, _ int) bool {
+		return knownValues.Has(val)
+	})
+
+	// If there are only invalid values, set an error to transition the nodepool's readiness to false
+	if len(values) == 0 {
+		return fmt.Errorf("no valid values found in %v for %s, expected one of: %v, got: %v",
+			requirement.Values, requirement.Key, knownValues, invalidValues)
+	}
+
+	// If there are valid and invalid values, log the invalid values and proceed with valid values
+	if len(invalidValues) > 0 {
+		log.FromContext(ctx).Error(fmt.Errorf("invalid values found for key"), "please correct found invalid values, proceeding with valid values", "key", requirement.Key, "valid-values", values, "invalid-values", invalidValues)
```

Comment:
nit: we can use a structured error here, same comment for the rest of the errors we create with associated values in the file.

Reply:
I'm not sure I follow, but I dropped all of these section anyway